pulsar-common/src/main/java/org/apache/pulsar/common/util/FutureUtil.java

```diff
@@ -258,4 +259,25 @@ public static CompletionException wrapToCompletionException(Throwable throwable)
             return new CompletionException(throwable);
         }
     }
+
+    /**
+     * Creates a new {@link CompletableFuture} instance catching
+     * potential exceptions and completing the future exceptionally.
+     *
+     * @param runnable the runnable to execute
+     * @param executor  the executor to use for executing the runnable
+     * @param completableFuture  the future to complete in case of exceptions
+     * @return
+     */
+
+    public static void safeRunAsync(Runnable runnable,
+                                    Executor executor,
+                                    CompletableFuture completableFuture) {
+        CompletableFuture
+                .runAsync(runnable, executor)
+                .exceptionally((throwable) -> {
+                    completableFuture.completeExceptionally(throwable);
```

Comment:
should we complete the Future on the same Executor ?
do we provide any guarantees on the thread that completes exceptionally the Future ?

Reply:
the exceptionally method body is very simple and without blocking operations. I don't think it's necessary